Addressed the requested items:\n- DONE: Ensured the guard short-circuit recreates the PID file before exiting ().\n- DONE: Removed dead helpers/imports in  and tightened the lambdas/fixtures.\n- DONE: Added docstrings plus the iterable-unpacking tweak in .\n- NOT DONE: Kept the port constant hard-coded at 10000 because the existing script header explicitly forbids changing that value for Codex compatibility.\n\nAll tests re-run with ........................................................................ [ 64%]
........................................                                 [100%]
=============================== warnings summary ===============================
tests/test_proxy.py:29
  /Users/jleechan/projects_other/codex_plus/tests/test_proxy.py:29: PytestUnknownMarkWarning: Unknown pytest.mark.slow - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.slow  # Mark as slow due to timeout issues in CI

tests/test_proxy.py:100
  /Users/jleechan/projects_other/codex_plus/tests/test_proxy.py:100: PytestUnknownMarkWarning: Unknown pytest.mark.slow - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.slow  # Mark as slow due to timeout issues in CI

tests/test_proxy.py: 8 warnings
tests/test_regression_ci_async_support.py: 1 warning
tests/test_request_logger.py: 1 warning
  /Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/site-packages/_pytest/unraisableexception.py:67: PytestUnraisableExceptionWarning: Exception ignored in: <function BaseSubprocessTransport.__del__ at 0x1076c9940>
  
  Traceback (most recent call last):
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/asyncio/base_subprocess.py", line 126, in __del__
      self.close()
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/asyncio/base_subprocess.py", line 104, in close
      proto.pipe.close()
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/asyncio/unix_events.py", line 566, in close
      self._close(None)
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/asyncio/unix_events.py", line 590, in _close
      self._loop.call_soon(self._call_connection_lost, exc)
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/asyncio/base_events.py", line 762, in call_soon
      self._check_closed()
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/asyncio/base_events.py", line 520, in _check_closed
      raise RuntimeError('Event loop is closed')
  RuntimeError: Event loop is closed
  
  Enable tracemalloc to get traceback where the object was allocated.
  See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))

tests/test_request_logger.py::TestRequestLogger::test_json_parsing_with_invalid_json
  /Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/site-packages/_pytest/unraisableexception.py:67: PytestUnraisableExceptionWarning: Exception ignored in: <function BaseSubprocessTransport.__del__ at 0x1076c9940>
  
  Traceback (most recent call last):
    File "/Users/jleechan/projects_other/codex_plus/src/codex_plus/request_logger.py", line 90, in _log_payload_to_file_async
      payload = json.loads(body)
                ^^^^^^^^^^^^^^^^
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/json/__init__.py", line 346, in loads
      return _default_decoder.decode(s)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/json/decoder.py", line 337, in decode
      obj, end = self.raw_decode(s, idx=_w(s, 0).end())
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/json/decoder.py", line 353, in raw_decode
      obj, end = self.scan_once(s, idx)
                 ^^^^^^^^^^^^^^^^^^^^^^
  json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 16 (char 15)
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/asyncio/base_subprocess.py", line 126, in __del__
      self.close()
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/asyncio/base_subprocess.py", line 104, in close
      proto.pipe.close()
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/asyncio/unix_events.py", line 566, in close
      self._close(None)
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/asyncio/unix_events.py", line 590, in _close
      self._loop.call_soon(self._call_connection_lost, exc)
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/asyncio/base_events.py", line 762, in call_soon
      self._check_closed()
    File "/Users/jleechan/.pyenv/versions/3.11.10/lib/python3.11/asyncio/base_events.py", line 520, in _check_closed
      raise RuntimeError('Event loop is closed')
  RuntimeError: Event loop is closed
  
  Enable tracemalloc to get traceback where the object was allocated.
  See https://docs.pytest.org/en/stable/how-to/capture-warnings.html#resource-warnings for more info.
    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
112 passed, 1 skipped, 13 warnings in 2.45s.

--- a/src/codex_plus/port_guard.py
+++ b/src/codex_plus/port_guard.py
@@ -1,3 +1,5 @@
+"""Utilities for determining whether the Codex proxy already owns a port."""
+
 from __future__ import annotations
 
 import argparse
@@ -27,6 +29,8 @@ class PortGuardError(RuntimeError):
 
 
 class PortState(str, Enum):
+    """Possible ownership states for a TCP port."""
+
     FREE = "free"
     OWNED_BY_PROXY = "owned_by_proxy"
     OCCUPIED_OTHER = "occupied_other"
@@ -35,16 +39,22 @@ class PortState(str, Enum):
 
 @dataclass(frozen=True)
 class ProcessInfo:
+    """Minimal representation of a process claimed by ``lsof``."""
+
     pid: int
     command: str
 
 
 @dataclass(frozen=True)
 class PortCheckResult:
+    """Outcome of a port ownership check."""
+
     state: PortState
     processes: tuple[ProcessInfo, ...]
 
     def to_dict(self) -> dict[str, object]:
+        """Render the result as a JSON-serialisable dictionary."""
+
         return {
             "state": self.state.value,
             "processes": [
@@ -55,7 +65,9 @@ class PortCheckResult:
 
 
 def _run_lsof(port: int) -> list[ProcessInfo]:
-    command = list(_LSOF_COMMAND) + [f"-iTCP:{port}", "-sTCP:LISTEN", "-Fpc"]
+    """Return listening processes for ``port`` using ``lsof`` in list format."""
+
+    command = [*_LSOF_COMMAND, f"-iTCP:{port}", "-sTCP:LISTEN", "-Fpc"]
     try:
         proc = subprocess.run(  # noqa: S603 - controlled command, arguments list
             command,
@@ -99,6 +111,8 @@ def _run_lsof(port: int) -> list[ProcessInfo]:
 
 
 def _probe_health(url: str, timeout: float = 1.0) -> bool:
+    """Check whether ``url`` responds with a 2xx status within ``timeout`` seconds."""
+
     try:
         with urllib.request.urlopen(url, timeout=timeout) as response:  # noqa: S310 - intentional
             if response.status >= 200 and response.status < 300:
@@ -109,6 +123,8 @@ def _probe_health(url: str, timeout: float = 1.0) -> bool:
 
 
 def _matches_expected(process: ProcessInfo, expected_markers: Iterable[str]) -> bool:
+    """Return ``True`` if the process command contains any of the expected markers."""
+
     haystack = process.command.lower()
     return any(marker.lower() in haystack for marker in expected_markers)
 
@@ -120,6 +136,8 @@ def check_port_ownership(
     health_url: str | None = None,
     health_timeout: float = 1.0,
 ) -> PortCheckResult:
+    """Determine who owns ``port`` using command markers and an optional health probe."""
+
     markers = expected_markers or _DEFAULT_EXPECTED_MARKERS
     try:
         processes = tuple(_run_lsof(port))
@@ -149,6 +167,8 @@ _EXIT_CODE_BY_STATE = {
 
 
 def _cli() -> int:
+    """Entry point for the ``python -m codex_plus.port_guard`` command."""
+
     parser = argparse.ArgumentParser(description="Inspect ownership of the Codex proxy port")
     parser.add_argument("--port", type=int, required=True)
     parser.add_argument("--expect", action="append", dest="expected", default=[])
@@ -176,6 +196,8 @@ def _cli() -> int:
 
 
 def main() -> None:  # pragma: no cover - simple wrapper
+    """Console script wrapper for the port guard CLI."""
+
     sys.exit(_cli())
 
 

--- a/tests/test_port_guard.py
+++ b/tests/test_port_guard.py
@@ -1,19 +1,9 @@
 from __future__ import annotations
 
-import types
-from typing import List
-
 import pytest
 
 from codex_plus import port_guard
 from codex_plus.port_guard import PortState, PortCheckResult
-
-
-class DummyCompletedProcess:
-    def __init__(self, stdout: str, returncode: int = 0, stderr: str = "") -> None:
-        self.stdout = stdout
-        self.returncode = returncode
-        self.stderr = stderr
 
 
 @pytest.fixture()
@@ -23,10 +13,7 @@ def reset_module_cache(monkeypatch: pytest.MonkeyPatch) -> None:
 
 
 def test_check_port_ownership_free_port(monkeypatch: pytest.MonkeyPatch) -> None:
-    def fake_run(*args, **kwargs):  # noqa: ANN001
-        return DummyCompletedProcess(stdout="")
-
-    monkeypatch.setattr(port_guard, "_run_lsof", lambda port: [])
+    monkeypatch.setattr(port_guard, "_run_lsof", lambda _port: [])
     result = port_guard.check_port_ownership(port=10000, expected_markers=("codex_plus",))
     assert result == PortCheckResult(state=PortState.FREE, processes=())
 
@@ -35,17 +22,18 @@ def test_check_port_ownership_owned_by_proxy_via_command(monkeypatch: pytest.Mon
     processes = [
         port_guard.ProcessInfo(pid=1234, command="python -c from codex_plus.main_sync_cffi import app"),
     ]
-    monkeypatch.setattr(port_guard, "_run_lsof", lambda port: processes)
+    monkeypatch.setattr(port_guard, "_run_lsof", lambda _port: processes)
     result = port_guard.check_port_ownership(port=10000, expected_markers=("codex_plus",))
     assert result.state is PortState.OWNED_BY_PROXY
     assert result.processes == tuple(processes)
 
 
 def test_check_port_ownership_owned_by_proxy_via_health(monkeypatch: pytest.MonkeyPatch) -> None:
-    monkeypatch.setattr(port_guard, "_run_lsof", lambda port: [port_guard.ProcessInfo(pid=4321, command="python other")])
+    monkeypatch.setattr(port_guard, "_run_lsof", lambda _port: [port_guard.ProcessInfo(pid=4321, command="python other")])
 
-    def fake_health(url: str, timeout: float) -> bool:  # noqa: ANN001
+    def fake_health(url: str, timeout: float) -> bool:
         assert url == "http://127.0.0.1:10000/health"
+        assert timeout == 1.0
         return True
 
     monkeypatch.setattr(port_guard, "_probe_health", fake_health)
@@ -58,16 +46,18 @@ def test_check_port_ownership_owned_by_proxy_via_health(monkeypatch: pytest.Monk
 
 
 def test_check_port_ownership_occupied(monkeypatch: pytest.MonkeyPatch) -> None:
-    monkeypatch.setattr(port_guard, "_run_lsof", lambda port: [port_guard.ProcessInfo(pid=77, command="redis-server")])
-    monkeypatch.setattr(port_guard, "_probe_health", lambda url, timeout: False)
+    monkeypatch.setattr(port_guard, "_run_lsof", lambda _port: [port_guard.ProcessInfo(pid=77, command="redis-server")])
+    monkeypatch.setattr(port_guard, "_probe_health", lambda _url, _timeout: False)
     result = port_guard.check_port_ownership(port=10000, expected_markers=("codex_plus",))
     assert result.state is PortState.OCCUPIED_OTHER
     assert result.processes[0].pid == 77
 
 
 def test_check_port_ownership_unknown_on_error(monkeypatch: pytest.MonkeyPatch) -> None:
-    def raising_lsof(port: int) -> List[port_guard.ProcessInfo]:  # noqa: ANN001
-        raise port_guard.PortGuardError("lsof missing")
+    error = port_guard.PortGuardError("lsof missing")
+
+    def raising_lsof(_port: int) -> tuple[port_guard.ProcessInfo, ...]:
+        raise error
 
     monkeypatch.setattr(port_guard, "_run_lsof", raising_lsof)
     result = port_guard.check_port_ownership(port=10000, expected_markers=("codex_plus",))

